backend/coreapp/compiler_wrapper.py

```diff
@@ -329,20 +338,15 @@ def filter_compiler_flags(compiler: str, compiler_flags: str) -> str:
         return " ".join(flags)
 
     @staticmethod
-    def compile_code(compiler: str, compiler_flags: str, code: str, context: str, to_regenerate: Optional[Compilation] = None) -> Tuple[Optional[Compilation], Optional[str]]:
+    @lru_cache(maxsize=settings.COMPILATION_CACHE_SIZE, typed=True)
```

Comment:
You probably don't need `typed=True`? All the args are always `str`?

From [`lru_cache` docs](https://docs.python.org/3/library/functools.html#functools.lru_cache)
> If typed is set to true, function arguments of different types will be cached separately. If typed is false, the implementation will usually regard them as equivalent calls and only cache a single result. (Some types such as str and int may be cached separately even when typed is false.)

Reply:
Ah, I figured it was just extra cautious, but yeah in practice it won't matter